test/test_mps.py

```diff
@@ -2356,6 +2356,12 @@ def helper(shape):
             self.assertEqual(y, ref_y)
             self.assertEqual(x.grad, cpu_x.grad)
 
+            # test in-place with storage_offset
+            y = x.detach()[1].tanh_()
+            ref_y = cpu_x.detach()[1].tanh_()
+
+            self.assertEqual(y, ref_y)
+
```

Comment:
This doesn't test non-contiguous cases but the LSTM (macOS 12) tests actually test them